### PR TITLE
feat: continual learning model and experiments

### PR DESCRIPTION
**Pull request by hlee9212**
Last large chunk of experiments for continual learning. Reviewing previous PRs first is recommended to understand the pattern of LightningModule and configuration for experiments.

### CODE REVIEW

File: pytorch/src/models/continual_vit_module.py [hunk 1]
```diff
@@ -0,0 +1,378 @@
...
+                - unit_quaternion: Ground truth rotation quaternions of shape (batch_size, 4)
+        """
+        rgbd_image, object_id, unit_quaternion = (
+            batch["rgbd_image"],
+            batch["object_id"],
+            batch["unit_quaternion"],
+        )
+        pred_class, pred_quaternion = self.forward(rgbd_image)
+
+        # Always use all_seen_classes for valid classes
```
Comment (nielsleadholm) [inline pytorch/src/models/continual_vit_module.py:124]: **suggestion:** Add a bit more info like, "This is a form of forward masking that ensures the model does not try to predict classes it has not seen yet."

File: pytorch/src/models/continual_vit_module.py [hunk 2]
```diff
@@ -0,0 +1,378 @@
...
+
+from typing import Optional
+
+import torch
+import torch.nn as nn
+import torch.optim as optim
+from torchmetrics import MeanMetric
+
+from src.losses.loss import masked_cross_entropy_loss, quaternion_geodesic_loss
+from src.metrics.continual_accuracy import calculate_accuracy
```
Comment (nielsleadholm) [inline pytorch/src/models/continual_vit_module.py:28]: **issue:** This looks like an invalid import. It should be called `calculate_continual_accuracy`.

File: pytorch/src/models/continual_vit_module.py [hunk 3]
```diff
@@ -0,0 +1,378 @@
...
+        """Log metrics for a given prefix (train, val, test).
+
+        Args:
+            prefix: Dataset split identifier ('train', 'val', or 'test')
+            loss: Combined loss tensor of shape (1,)
+            classification_loss: Classification loss component of shape (1,)
+            quaternion_geodesic_loss: Quaternion loss component of shape (1,)
+            pred_class: Class prediction logits of shape (batch_size, num_classes)
+            pred_quaternion: Quaternion prediction tensor of shape (batch_size, 4)
+            object_id: Ground truth object class IDs of shape (batch_size,)
```
Comment (nielsleadholm) [inline pytorch/src/models/continual_vit_module.py:168]: **nitpick:** `object_ids` with an s would be clearer, this makes it seem like there will only be one class.

File: pytorch/src/eval_continual.py [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+    os.makedirs(cfg.save_dir, exist_ok=True)
+
+    # Initialize and validate components
+    datamodule, model = _setup_components(cfg)
+    _validate_classes(datamodule, model)
+
+    # Setup evaluation
+    trainer = setup_trainer(cfg)
+    predictions = generate_predictions(trainer, model, datamodule, ckpt_path)
+    masker = _get_masked_predictions_fn(model_id)
```
Comment (nielsleadholm) [inline pytorch/src/eval_continual.py:84]: **question:** Is this also dealing with quaternion predictions? In particular, `ycb_continual_datamodule` does not break up the evaluation data into different tasks. Now we don't need to mask the _outputs_ of quaternions (it is always the case that the model should output the 4 neurons for the four quaternion values), however we don't want to evaluate rotation error on objects that the ViT has not been trained on. As such, we should not be feeding in objects that have never been trained on, and getting a rotation error as a result (either for loss during training, or for metrics during evaluation). I'm unclear where this is being controlled for in the case of validation.

**Review by nielsleadholm — COMMENTED** [review]
Overall looks really nice, thanks for the well structured code Hojae! Just a few clarifications.

**Review by nielsleadholm — APPROVED** [review]
Thanks for those updates, looks good to me!
